evalml/tests/automl_tests/test_automl.py

```diff
@@ -4488,3 +4702,72 @@ def test_automl_baseline_pipeline_predictions_and_scores_time_series(problem_typ
     np.testing.assert_allclose(
         baseline.feature_importance.iloc[:, 1], np.array([0.0] * X.shape[1])
     )
+
+
+@pytest.mark.parametrize(
+    "objective,errors",
+    [
+        ("Log Loss Binary", True),
+        ("AUC", True),
+        ("F1", False),
+        ("Accuracy Binary", False),
+    ],
+)
+@patch(
+    "evalml.objectives.binary_classification_objective.BinaryClassificationObjective.optimize_threshold",
+    return_value=0.65,
+)
+def test_automl_alternate_thresholding_objective(
+    mock_optimize, objective, errors, X_y_binary
+):
+    X, y = X_y_binary
+    if errors:
+        with pytest.raises(
+            ValueError,
+            match="Alternate thresholding objective must be a tuneable objective",
+        ):
+            AutoMLSearch(
+                X_train=X,
+                y_train=y,
+                problem_type="binary",
+                alternate_thresholding_objective=objective,
+            )
+        return
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        alternate_thresholding_objective=objective,
+        max_iterations=1,
+    )
+    automl.search()
```

Comment:
Following up on [outdated comment](https://github.com/alteryx/evalml/pull/2320#discussion_r646985865). @bchen1116 said:
> @dsherry It takes about 4 seconds to run. I can set max_iterations even lower and get it to run in under 2 seconds. This is comparable to the runtime when we mock, but with this, and the following test, I wanted to make sure the thresholds were being set properly. Would it still be preferable to mock the fit, score, predict_proba, and other methods here?

Oh lol so with `max_iterations=1` this is setting the threshold on the baseline pipeline. Sure that's fine. Just as long as we aren't burning test runtime training real models when we don't need to for this test!